fuel/datasets/hdf5.py

```diff
@@ -411,21 +620,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
+            raise ValueError()
         data = []
         shapes = []
         handle = self._file_handle
         for source_name, subset in zip(self.sources, self.subsets):
-            if isinstance(request, slice):
-                req = slice(request.start + subset.start,
-                            request.stop + subset.start, request.step)
+            if hasattr(subset, 'step'):
```

Comment:
Thanks! I am a bit surprised though that you chose the `hasattr` variant. May  I ask why you like it more than the `isinstance` one?